tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/XUnit/V3/XUnitTestMethodRunnerBaseRunTestCaseV3Integration.cs

```diff
@@ -0,0 +1,243 @@
+            if (_runSummaryFieldCount != 5)
+            {
+                Common.Log.Warning("RunSummary type doesn't have the field count we are expecting. Flushing messages from RetryMessageBus");
+                retryMessageBus.FlushMessages(retryMetadata.UniqueID);
+                return returnValue;
+            }
+
+            var runSummaryUnsafe = Unsafe.As<TReturn, RunSummaryUnsafeStruct>(ref returnValue);
+
+            var isFlakyRetryEnabled = retryMetadata.FlakyRetryEnabled;
+            var index = retryMetadata.ExecutionIndex;
+
+            if (index == 0)
+            {
+                // Let's make decisions based on the first execution regarding slow tests or retry failed test feature
+                if (isFlakyRetryEnabled)
+                {
+                    retryMetadata.TotalExecutions = CIVisibility.Settings.FlakyRetryCount + 1;
+                }
+                else
+                {
+                    var duration = TimeSpan.FromSeconds((double)runSummaryUnsafe.Time);
+                    retryMetadata.TotalExecutions = Common.GetNumberOfExecutionsForDuration(duration);
+                }
+
+                retryMetadata.ExecutionNumber = retryMetadata.TotalExecutions - 1;
+            }
+
+            if (retryMetadata.ExecutionNumber > 0)
+            {
+                var doRetry = true;
+                if (isFlakyRetryEnabled)
+                {
+                    var remainingTotalRetries = Interlocked.Decrement(ref _totalRetries);
+                    if (runSummaryUnsafe.Failed == 0)
+                    {
+                        Common.Log.Debug("EFD/Retry: [FlakyRetryEnabled] A non failed test execution was detected, skipping the remaining executions.");
+                        doRetry = false;
+                    }
+                    else if (runSummaryUnsafe.NotRun > 0)
+                    {
+                        Common.Log.Debug("EFD/Retry: [FlakyRetryEnabled] A NotRun test was detected, skipping the remaining executions.");
+                        doRetry = false;
+                    }
+                    else if (remainingTotalRetries < 1)
+                    {
+                        Common.Log.Debug<int>("EFD/Retry: [FlakyRetryEnabled] Exceeded number of total retries. [{Number}]", CIVisibility.Settings.TotalFlakyRetryCount);
+                        doRetry = false;
+                    }
+                }
+
+                if (doRetry)
+                {
+                    var retryNumber = retryMetadata.ExecutionIndex + 1;
+                    // Set the retry as a continuation of this execution. This will be executing recursively until the execution count is 0/
+                    Common.Log.Debug<int, int>("EFD/Retry: [Retry {Num}] Test class runner is duck casted, running a retry. [Current retry value is {Value}]", retryNumber, retryMetadata.ExecutionNumber);
+
+                    var mrunner = instance.DuckCast<IXunitTestMethodRunnerV3>()!;
```

Comment:
kinda strange you need the `!` here given you bail out at the start of the method, and the type isn't marked null anyway 😕

Reply:
fixed in https://github.com/DataDog/dd-trace-dotnet/pull/6573/commits/0bb1ed8bf288df45f34491c03669193dd5f59b8b